File: kotlin/java-to-kotlin/src/main/kotlin/section02/lec05/Lec05Main.java
package section02.lec05;

import java.util.Arrays;
import java.util.List;

public class Lec05Main {

    public static void main(String[] args) {
        List<Long> numbers = Arrays.asList(1L, 2L, 3L);

        for (long number: numbers) {
            System.out.println(number);
        }

        for (int i = 1; i <= 3; i++) {
            System.out.println(i);
        }

        for (int i = 3; i >= 1; i--) {
            System.out.println(i);
        }
    }

}

File: kotlin/java-to-kotlin/src/main/kotlin/section02/lec05/Lec05Main.kt
package section02.lec05

fun main() {

}

fun validateScoreIsNotNegative(score: Int) {
    if (score in 0..100) {
        throw IllegalArgumentException("score 의 범위는 0부터 100입니다.")
    }
}

fun getPassOrFail(score: Int): String {
    return if (score >= 50) {
        "P"
    } else {
        "F"
    }
}

fun getGrade(score: Int): String {
    return if (score >= 90) {
        "A"
    } else if (score >= 80) {
        "B"
    } else if (score >= 80) {
        "C"
    } else {
        "D"
    }
}

fun getGradeWithSwitch(score: Int): String {
//    return when (score / 10) {
//        9 -> "A"
//        8 -> "B"
//        7 -> "C"
//        else -> "D"
//    }
    return when (score) {
        in 90..99 -> "A"
        in 80..89 -> "B"
        in 70..79 -> "C"
        else -> "D"
    }
}

fun judgeNumber(number: Int): String {
    return when (number) {
        -1, 1, 0 -> "1"
        else -> "2"
    }
}

fun judgeNumber2(number: Int) {
    when {
        number == 0 -> println("1")
        number % 2 == 0 -> println("2")
        else -> println("3")
    }
}